backend/infrahub/core/utils.py

```diff
@@ -191,6 +191,12 @@ def convert_ip_to_binary_str(
     return ip_bin.zfill(obj.max_prefixlen)
 
 
+def collapse_ipv6_address_or_network(address_or_network: str) -> str:
+    if "/" in address_or_network:
+        return ipaddress.IPv6Network(address_or_network).with_prefixlen
+    return str(ipaddress.IPv6Address(address_or_network))
```

Comment:
The issue doesn't specifically mention the search ahead aspect of the search anywhere. But one thing to note is that the above function would only work if we have a valid ip address. So for instance if the user types in `"2001:0db8:0"` this is not a valid address in itself however the intention might be to search any address matching this start. With this in mind it could be that the string `"2001:0db8:0"` should be converted to -> `"2001:db8:"` before searching the database. If we expect users to always enter the full ip addresses this wouldn't be an issue but I'd assume that the expectation would be that you can start off typing a few characters. Alternatively we could have a quickfix first to make it possible to do a full search to begin with and then expand upon that.

Reply:
Right, I commited changes to support that.